### PR TITLE
kill report pillows

### PR DESCRIPTION
**Pull request by snopoke**
@calellowitz 

cc @czue

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/reindexer/change_providers/form.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from corehq.apps.domain.models import Domain
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/reindexer/change_providers/form.py]: F401 'Domain' imported but unused

File: corehq/ex-submodules/pillowtop/tests/test_form_change_providers.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import itertools
+import uuid
+from unittest import TestCase
+
+from django.test import SimpleTestCase
+from fakecouch import FakeCouchDb
+
+from corehq.form_processor.backends.sql.dbaccessors import doc_type_to_state, FormAccessorSQL
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/tests/test_form_change_providers.py]: F401 'FormAccessorSQL' imported but unused

File: corehq/util/doc_processor/couch.py [hunk 3]
```diff
@@ -1,36 +1,10 @@
 from couchdbkit import ResourceNotFound
 
-from corehq.util.couch_helpers import PaginatedViewArgsProvider
+from corehq.util.couch_helpers import PaginatedViewArgsProvider, MultiKeyViewArgsProvider
```
Comment (dimagimon) [inline corehq/util/doc_processor/couch.py]: F401 'PaginatedViewArgsProvider' imported but unused

File: corehq/util/pagination.py [hunk 4]
```diff
@@ -61,6 +61,29 @@ def get_next_args(self, last_item, *last_args, **last_kwargs):
...
+    """
+    def __init__(self, kwargs_list):
+        self.kwargs_list = list(kwargs_list)
+
+    def get_initial_args(self):
+        return [], self.kwargs_list[0]
+
+    def get_next_args(self, result, *last_args, **last_kwargs):
+        # all docs for the current key have been processed
+        # move on to the next key combo
```
Comment (czue) [inline corehq/util/pagination.py]: should this comment be here?

File: corehq/ex-submodules/pillowtop/reindexer/reindexer.py [hunk 5]
```diff
@@ -61,7 +61,10 @@ def consume_options(self, options):
 
     def reindex(self):
         for change in self.change_provider.iter_all_changes(start_from=self.start_from):
-            self.pillow.process_change(change)
+            try:
+                self.pillow.process_change(change)
+            except Exception:
+                pillow_logging.exception("Uable to process change: %s", change.id)
```
Comment (czue) [inline corehq/ex-submodules/pillowtop/reindexer/reindexer.py]: nit: "uable"

**Comment by czue** [comment]
looks great. is this ready to be merged?

**Comment by snopoke** [comment]
I think this is GTG. I've checked the kafka offsets and the new pillows are both caught up.

**Comment by czue** [comment]
test fail might be real but i kicked it:

```
======================================================================
ERROR: test suite for <class 'pillowtop.tests.test_form_change_providers.TestSqlDomainFormChangeProvider'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "corehq/ex-submodules/pillowtop/tests/test_form_change_providers.py", line 71, in setUpClass
    for domain in cls.domains
  File "corehq/ex-submodules/pillowtop/tests/test_form_change_providers.py", line 72, in <dictcomp>
    for doc_type in ['XFormInstance', 'XFormArchived']
  File "corehq/ex-submodules/pillowtop/tests/test_form_change_providers.py", line 62, in _create_form
    form = create_form_for_test(domain, state=doc_type_to_state[doc_type])
  File "/mnt/commcare-hq-ro/corehq/form_processor/tests/utils.py", line 235, in create_form_for_test
    FormProcessorSQL.save_processed_models(ProcessedForms(form, None), cases)
  File "/mnt/commcare-hq-ro/corehq/form_processor/backends/sql/processor.py", line 73, in save_processed_models
    with transaction.atomic():
  File "/vendor/lib/python2.7/site-packages/django/db/transaction.py", line 260, in __enter__
    if not connection.get_autocommit():
  File "/vendor/lib/python2.7/site-packages/django/db/backends/__init__.py", line 345, in get_autocommit
    self.ensure_connection()
  File "/vendor/lib/python2.7/site-packages/django/db/backends/__init__.py", line 133, in ensure_connection
    self.connect()
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/__init__.py", line 133, in ensure_connection
    self.connect()
  File "/vendor/lib/python2.7/site-packages/django/db/backends/__init__.py", line 122, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 130, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/vendor/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
OperationalError: FATAL:  database "test_commcarehq" does not exist
```

**Comment by czue** [comment]
(seems to be failing consistently)

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by czue** [comment]
:tada:

**Pull request merged**